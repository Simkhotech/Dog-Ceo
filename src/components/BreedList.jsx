import React from 'react';
import PropTypes from 'prop-types';
import { Typeahead } from 'react-bootstrap-typeahead';
import Panel from 'react-bootstrap/lib/Panel';
import Col from 'react-bootstrap/lib/Col';

import Alert from './Alert';

export default class BreedList extends React.Component {
  componentDidMount() {
    this.props.getBreedList();
  }

  render() {
    const {
      breedList,
      hasError,
      message,
    } = this.props;
    const title = 'Выберите породу...';

    function getDogPageUrl(selected) {
      return `/dog/${selected.breed}${selected.subBreed ? `/${selected.subBreed}` : ''}`;
    }

    if (hasError) {
      return (<Alert message={message} />);
    }

    return (
      <Panel>
        <Panel.Heading>
          <Panel.Title componentClass="h3">{title}</Panel.Title>
        </Panel.Heading>
        <Panel.Body>

          <Typeahead
            onChange={(selected) => {
              if (selected.length > 0) {
                window.location = getDogPageUrl(selected[0]);
              }
            }}
            options={breedList}
          />

        </Panel.Body>
      </Panel>
    );
  }
}

BreedList.propTypes = {
  getBreedList: PropTypes.func.isRequired,
  breedList: PropTypes.arrayOf(PropTypes.object).isRequired,
  hasError: PropTypes.bool,
  message: PropTypes.string,
};

BreedList.defaultProps = {
  hasError: false,
  message: null,
};
