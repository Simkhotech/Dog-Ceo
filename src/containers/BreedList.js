import { connect } from 'react-redux';
import getBreedList from '../actions/breedList';
import BreedList from '../components/BreedList';

const mapStateToProps = state => ({
  breedList: state.breedList.breedList.breedList,
  hasError: state.breedList.services.breedList.hasError,
  message: state.breedList.services.breedList.message,
});

const mapDispatchToProps = { getBreedList };

export default connect(
  mapStateToProps,
  mapDispatchToProps,
)(BreedList);
