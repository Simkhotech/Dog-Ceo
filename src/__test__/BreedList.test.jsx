/* eslint-env jest */
import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';

import BreedList from '../components/App';

// Important!
// I do not know JEST and I think
// it will be useful to study it.
// The tests were not so easy for
// me to learn. Need more time
describe('BreedList suite', () => {
  it('renders without crashing... so good test!', () => {
    const div = document.createElement('div');
    ReactDOM.render(<BreedList />, div);
    ReactDOM.unmountComponentAtNode(div);
  });

  it('404 renders correctly with snapshot', () => {
    const tree = renderer
      .create(<BreedList />)
      .toJSON();

    expect(tree).toMatchSnapshot();
  });
});
