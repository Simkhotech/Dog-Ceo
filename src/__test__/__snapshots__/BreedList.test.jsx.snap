// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BreedList suite 404 renders correctly with snapshot 1`] = `
Array [
  <nav
    className="navbar navbar-default"
  >
    <div
      className="container"
    >
      <div
        className="navbar-header"
      >
        <a
          className="navbar-brand"
          href="/"
        >
          Dog-ceo
        </a>
      </div>
    </div>
  </nav>,
  <div
    className="alert alert-info"
    role="alert"
  >
    <strong>
      404
    </strong>
     
    There does not seem to be a page here
  </div>,
]
`;
